src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -214,14 +220,12 @@ void Server::handleEvent(JNIEnv* env, const inotify_event* event) {
     } else if (IS_SET(mask, IN_MODIFY)) {
         type = MODIFIED;
     } else {
-        type = UNKNOWN;
+        logToJava(WARNING, "Unknown event 0x%x for %s", mask, utf16ToUtf8String(path).c_str());
+        reportUnknownEvent(env, path);
```

Comment:
We could report the unknown event flags to the Java side for all OSes, so we wouldn't need to dig them up from the log. Though it is probably a non issue.